python/dgl/mock_sparse2/sddmm.py

```diff
@@ -0,0 +1,54 @@
+"""Sampled Dense-Dense Matrix Multiplication (SDDMM) operator module."""
+import torch
+
+from .sparse_matrix import SparseMatrix
+
+__all__ = ["sddmm"]
+
+
+def sddmm(
+    A: SparseMatrix, mat1: torch.Tensor, mat2: torch.Tensor
+) -> SparseMatrix:
+    r"""Sampled-Dense-Dense Matrix Multiplication (SDDMM).
+
+    ``sddmm`` multiplies two dense matrices :attr:``mat1`` and :attr:``mat2``
+    at the nonzero locations of sparse matrix :attr:``A``. Values of :attr:``A``
+    is not considered during the computation.
+
+    Mathematically ``sddmm`` is formulated as:
+
+    .. math::
+        out = (mat1 @ mat2) * A
+
+    Parameters
+    ----------
+    A : SparseMatrix
+        Sparse matrix of shape `(M, N)`.
+    mat1 : Tensor
+        Dense matrix of shape `(M, K)`
+    mat2 : Tensor
+        Dense matrix of shape `(K, N)`
```

Comment:
From the test cases, it seems that you also support `mat1` and `mat2` being vectors? If so, you may want to extend the shape description here and the example section too.

Reply:
Feel free to fix it in another PR if you want to unblock the development of spspmm.

Reply:
OK. Will fix it in the SpSpMM PR.